code/modules/mining/mine_items.dm

```diff
@@ -428,3 +431,21 @@
 	anchored = 1
 	layer = MOB_LAYER - 0.2
 	density = 0
+
+/obj/item/weapon/emptysandbag
+	name = "empty sandbag"
+	desc = "A bag to be filled with sand."
+	icon = 'icons/obj/items.dmi'
+	icon_state = "sandbag"
+	w_class = 1
+
+/obj/item/weapon/emptysandbag/attackby(obj/item/W, mob/user, params)
+	if(istype(W,/obj/item/weapon/ore/glass || /obj/item/weapon/ore/glass/basalt))
```

Comment:
This is why it compiled and worked with both:
1. it tries to get the args for the first istype()
2. it gets ```W``` as one argument, and the STATEMENT ```/obj/item/weapon/ore/glass || /obj/item/weapon/ore/glass/basalt``` as the second argument
3. path || path2 is just path
4. so istype is now ```istype(W, /obj/item/weapon/ore/glass)```
5. ```/basalt``` is a subtype of ```obj/item/weapon/ore/glass``` so it's equivalent to the intended function but only because the second path is a subtype of the first.